### ISSUE TITLE
Proposal for New Provider Types - "Monitor" and "Test"

### ISSUE THREAD

**Issue by StewartW** [awslabs/aws-deployment-framework#391]
### What and Why
ADF currently has three different type of providers: Source, Build, Deploy. 

I'd like to propose two new types of providers for the default pipeline type. 
These are **Monitor** and **Test**
The purpose of these providers is to make it easier for customers to build in automated monitoring and testing into their pipelines. 

### How?
The new providers should each come with a default action
**Monitor**
A new provider that creates an action that happens after the deploy action has been ran for a target. 
The first version of this action should allow for a list of CloudWatch Alarm names to be specified. It will then monitor these alarms for a defined period of time. Once that time has passed, it will then progress the pipeline to the next stage/action as appropriate. 

**Test**
A new provider that creates an action that happens after the deploy action has been ran for a target, but can either run in parallel or before the monitor stage. 
The first version of this action should allow for basic HTTP/S endpoints to be invoked with an expected response code checked. 
If it passes, it will then progress the pipeline to next action or if it fails, it will stop the pipeline. 

### TL;DR
Two new providers that allow for automated testing and monitoring built into the pipeline.

**Comment by sbkok** [comment]
I'm closing this issue as it has been inactive for a long time. We have decided not to implement this feature, due to lack of interest from other customers at this time. Please reopen if you think we should reconsider this.
Thank you!
